erts/emulator/sys/win32/sys_time.c

```diff
@@ -141,15 +141,27 @@ SystemTime2MilliSec(SYSTEMTIME *stp)
     return stime;
 }
 
+#define SKIP 0x3FF
+
 static ErtsMonotonicTime
 os_monotonic_time_qpc(void)
 {
     LARGE_INTEGER pc;
 
-    if (!(*internal_state.r.o.pQueryPerformanceCounter)(&pc))
-	erts_exit(ERTS_ABORT_EXIT, "QueryPerformanceCounter() failed\n");
+    ErtsMonotonicTime temp;
 
-    return (ErtsMonotonicTime) pc.QuadPart;
+    /*
+     * Windows QPC does not ensure that the returned clocks
+     * are monotonic over several threads, reduce resolution to ensure
+     * that we return monotonic clocks even on multiple threads.
+     */
+    do {
+        if (!(*internal_state.r.o.pQueryPerformanceCounter)(&pc))
+            erts_exit(ERTS_ABORT_EXIT, "QueryPerformanceCounter() failed\n");
+        temp = (ErtsMonotonicTime) pc.QuadPart;
+    } while(!(temp & SKIP));
+
+    return temp & (0xFFFFFFFFFFFFFFFF-SKIP);
```

Comment:
Use the `ERTS_I64_LITERAL()` macro on large integer literals, so you get the correct literal suffix for the specific compiler. That is `ERTS_I64_LITERAL(0xFFFFFFFFFFFFFFFF)` in this case.